template/template_test.go

```diff
@@ -112,6 +115,88 @@ func TestGenerator(t *testing.T) {
 	}
 }
 
+type k8sWrapper struct {
+	runtime.Object
+}
+
+func (w *k8sWrapper) MarshalJSON() ([]byte, error) {
+	return json.Marshal(w.Object)
+}
+
+func TestGeneratorWithKubernetesResource(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	gen := NewGenerator("./testassets/pods")
+	err := gen.CompileAndValidate()
+
+	g.Expect(err).To(Not(HaveOccurred()))
+
+	_, err = gen.RenderJSON()
+	g.Expect(err).To(Not(HaveOccurred()))
+
+	{
+		_, err = gen.WithResource(&k8sWrapper{Object: makePod()})
+
+		g.Expect(err).To(Not(HaveOccurred()))
+	}
+
+	{
+		pod := makePod()
```

Comment:
without `k8sWrapper`, CUE uses `reflect`:  https://github.com/cue-lang/cue/blob/9dbb373d9cb6aeddcd73981d0f79b637067ca517/internal/core/convert/go.go#L370

this is the paths that appears to be buggy.